Adiciona Dockerfile

joseemds opened the pull request:
- Adiciona Dockerfile para distribuição
- Simplifica algumas instruções do Makefile

lambda-loop committed:
some modifications in generate.php and the new dockerfile

lambda-loop committed:
instructions to run the dockerfile

joseemds committed:
chore: correct Makefile rules

joseemds committed:
chore: add exported files to .gitignore

joseemds committed:
docker: simplify Dockerfile

dadn2002 reviewed (dismissed):
Dificuldade quanto setup do ambiente, não contém instalação de dependências, script de autosetup/config.

Problema também quanto execução, localhost:8080 não responsivo após passos do dockereadme.org

[log.zip](https://github.com/user-attachments/files/19645547/log.zip)

dadn2002 commented:
É complicado conseguir executar esses passos em novos ambientes tendo que os refazer manualmente (A exemplo do git actions pra automação de testes).

Também o refatoramento do projeto como um todo que quebrou a antiga forma de inicialização do servidor.

joseemds committed:
fix Dockerfile sat path

dadn2002 reviewed (commented):
os passos do dockereadme.org iniciam o docker, porém o make start-server acusa:
`sh: 1: -s: not found`
mesmo problema da variável de ambiente não sendo identificada, mas isso é o de menos.

Commit:
fix: makefile not starting local server and many minor changes

Commit:
fix: makefile not starting local server and many minor changes

Commit:
fix: makefile not starting local server and many minor changes

Commit:
fix: makefile not starting local server and many minor changes

Commit:
feat: githook pre-commit added

make clean-all always execute before proper commit, removing subproducts and unwanted files.

Commit:
test: githook testing
